kdc/kerberos5.c

```diff
@@ -556,6 +556,12 @@ pa_pkinit_validate(astgs_request_t r, const PA_DATA *pa)
     kdc_audit_setkv_number((kdc_request_t)r, KDC_REQUEST_KV_AUTH_EVENT,
 			   KDC_AUTH_EVENT_PREAUTH_SUCCEEDED);
 
+    /*
+     * Match Windows by preferring the authenticator nonce over the one in the
+     * request body.
+     */
+    r->ek.nonce = _kdc_pk_nonce(pkp);
```

Comment:
Does this break interop with earlier versions of Heimdal?

Reply:
I don’t know. At any rate, we haven’t noticed this KDC‐side change cause any problems interoperating with the Heimdal client.

Reply:
Meaning that you deployed this fix on the KDCs without also upgrading all clients?